eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -155,12 +155,13 @@ jobs:
           jdkVersionOption: '1.11'
           jdkArchitectureOption: 'x64'
           publishJUnitResults: false
-          goals: 'spotbugs:check checkstyle:checkstyle-aggregate'
+          goals: 'spotbugs:check checkstyle:checkstyle-aggregate javadoc:aggregate'
```

Comment:
You'll have to excuse my naivete but what does this javadoc aggregate actually buy us here? Wouldn't any javadoc problems with a given module be found during the javadoc step in the build?

Reply:
This is validating that all JavaDocs for all client libraries remain valid. This helps to prevent broken references across libraries. The build step is only building the libraries for one service, so we may miss some changes.

Reply:
In a world where we have a pipeline per service, is validating all java docs still the correct thing though? For example, we might actually be building KeyVault, with a binary dependency on Core 1.0.0, but the source contains Core 1.1.0 (substitute versions as necessary to consider a future where the focus has gone off core and is on something else).

Sure ... its validating that the JavaDocs are consistent with what is in the repository, but thats not actually the that it is being built against.